fmt/src/git.rs

```diff
@@ -151,7 +153,7 @@ pub fn resolve_file_attrs(
 
                 Ok::<_, Infallible>(Default::default())
             })?;
-        prev_commit = this_commit;
+        next_commit = this_commit;
```

Comment:
This patch is a follow-up to https://github.com/korandoru/hawkeye/pull/150 and https://github.com/GitoxideLabs/gitoxide/discussions/1395.

@Byron I'm figuring out how to walk the final commit tree. That said, the current logic iterate all the commits and calculate the changes between a next commit and a current commit. The iteration will exit on the very first (root) commit, then all the changes in the root commit won't be handled.

I'm not quite sure how to get a full change list (or just iterate all the files in the root commit). Investigating...

Reply:
The idea is to have an empty source tree to compare with if a commit has no parents.
There is an alternative API now, https://docs.rs/gix/latest/gix/struct.Repository.html#method.diff_tree_to_tree, which might make this more obvious and/or easier to use.

Reply:
@Byron Thanks a lot for your information! I use it in https://github.com/korandoru/hawkeye/pull/173/commits/272eee03593e76b406d0e9023debae581c2e9617 and everything looks correct now.

It was the right choice to ask you first :D